src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -492,50 +492,58 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             var hasErrors = localSymbol.ScopeBinder
                 .ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
-            BoundBlock block;
-            if (node.Body != null)
-            {
-                block = BindEmbeddedBlock(node.Body, diagnostics);
-            }
-            else if (node.ExpressionBody != null)
+            BoundBlock blockBody = node.Body != null ? BindEmbeddedBlock(node.Body, diagnostics) : null;
+            DiagnosticBag expressionBodyDiagnostics = null;
+            BoundBlock expressionBody = null;
+            if (node.ExpressionBody != null)
             {
-                block = BindExpressionBodyAsBlock(node.ExpressionBody, diagnostics);
+                expressionBodyDiagnostics = node.Body == null ? diagnostics : new DiagnosticBag();
```

Comment:
>ode.Body == null [](start = 45, length = 16)

It looks like we are checking this condition and inlining it multiple times. It doesn't feel like this makes the code clearer, I would say quite the opposite. Consider refactoring into regular ```if``` statements. For example:
```
            BoundBlock blockBody = null;
            BoundBlock expressionBody = null;

            if (node.Body != null)
            {
                 blockBody = runAnalysis(BindEmbeddedBlock(node.Body, diagnostics), diagnostics);

                 if (node.ExpressionBody != null)
                 {
                     DIagnosticBag expressionBodyDiagnostics = new DiagnosticBag();
                     expressionBody = runAnalysis(BindExpressionBodyAsBlock(node.ExpressionBody, expressionBodyDiagnostics), expressionBodyDiagnostics);
                 }
            }
            else if (node.ExpressionBody != null)
            {
                 expressionBody = runAnalysis(BindExpressionBodyAsBlock(node.ExpressionBody, diagnostics), diagnostics);
            }
            else 
            {
                hasErrors = true;
                diagnostics.Add(ErrorCode.ERR_LocalFunctionMissingBody, localSymbol.Locations[0], localSymbol);
            }

            BoundBlock runAnalysis(BoundBlock block, DiagnosticBag blockDiagnostics)
            {
                if (block != null)
                {
                    // Have to do ControlFlowPass here because in MethodCompiler, we don't call this for synthed methods
                    // rather we go directly to LowerBodyOrInitializer, which skips over flow analysis (which is in CompileMethod)
                    // (the same thing - calling ControlFlowPass.Analyze in the lowering - is done for lambdas)
                    // It's a bit of code duplication, but refactoring would make things worse.
                    var endIsReachable = ControlFlowPass.Analyze(localSymbol.DeclaringCompilation, localSymbol, block, blockDiagnostics);
                    if (endIsReachable)
                    {
                        if (ImplicitReturnIsOkay(localSymbol))
                        {
                            block = FlowAnalysisPass.AppendImplicitReturn(block, localSymbol);
                        }
                        else
                        {
                            blockDiagnostics.Add(ErrorCode.ERR_ReturnExpected, localSymbol.Locations[0], localSymbol);
                        }
                    }
                }
                return block;
            }
```